packages/waku/src/lib/renderers/rsc-renderer.ts

```diff
@@ -188,7 +198,46 @@ export async function renderRsc(
     ) {
       // XXX This doesn't support streaming unlike busboy
       const formData = parseFormData(bodyStr, contentType);
-      args = await decodeReply(formData);
+      const moduleMap = new Proxy({} as Record<string, ImportManifestEntry>, {
+        get(_target, rsfId: string): ImportManifestEntry {
+          const [fileId, name] = rsfId.split('#') as [string, string];
+          // fixme: race condition, server actions are not initialized in the first time
```

Comment:
As said, I think we should initialize all server actions before running RSC otherwise it will crash for the first time

Reply:
for next.js, they have a global server action map in the runtime

https://github.com/vercel/next.js/blob/86dbcbea431f2fd85d94a72a9c2d35781154a5d7/packages/next/src/server/app-render/encryption-utils.ts#L138-L158

Reply:
I still need to understand the issue, but it feels like we want lazy evaluation.